plugins/org.eclipse.n4js/src/org/eclipse/n4js/organize/imports/ImportSpecifiersUtil.xtend

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright (c) 2017 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * 
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.organize.imports
+
+import java.util.List
+import org.eclipse.n4js.n4JS.ImportDeclaration
+import org.eclipse.n4js.n4JS.ImportSpecifier
+import org.eclipse.n4js.n4JS.NamedImportSpecifier
+import org.eclipse.n4js.n4JS.NamespaceImportSpecifier
+import org.eclipse.n4js.ts.types.TExportableElement
+
+/**
+ * Utilities for ImportSpecifiers
+ */
+class ImportSpecifiersUtil {
+
+	/**
+	 * @return {@link List} of {@link ImportProvidedElement}s describing imported elements
+	 */
+	public static def List<ImportProvidedElement> mapToImportProvidedElements(List<ImportSpecifier> importSpecifiers) {
+		importSpecifiers.map(
+			specifier |
+				switch (specifier) {
+					NamespaceImportSpecifier:
+						if (specifier.importedModule !== null) {
+							val importProvidedElements = newArrayList
+							// add import provided element for a namespace itself
+							importProvidedElements.add(
+								new ImportProvidedElement(specifier.alias, computeNamespaceActualName(specifier),
+									specifier))
```

Comment:
_And it's all in one return-statement._ Did you paste/copy this? It's really difficult.

Reply:
I am afraid I wrote it, years ago when I was fascinated with expressiveness of Xtend.
Refactored.